aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/states/AuthenticationState.kt

```diff
@@ -100,18 +113,51 @@ sealed class AuthenticationState : State {
                         }
                     }
                 }
-                is SignedOut -> when (authenticationEvent) {
-                    is AuthenticationEvent.EventType.SignInRequested -> {
-                        val action = authenticationActions.initiateSignInAction(authenticationEvent)
-                        StateResolution(SigningIn(), listOf(action))
+                is SignedOut -> {
+                    val authorizationEvent = event.isAuthorizationEvent()
+                    when {
+                        authenticationEvent is AuthenticationEvent.EventType.SignInRequested -> {
+                            val action = authenticationActions.initiateSignInAction(authenticationEvent)
+                            StateResolution(SigningIn(), listOf(action))
+                        }
+                        authenticationEvent is AuthenticationEvent.EventType.SignOutRequested -> {
+                            val action = authenticationActions.initiateSignOutAction(authenticationEvent, null)
+                            StateResolution(SigningOut(), listOf(action))
+                        }
+                        authorizationEvent is AuthorizationEvent.EventType.StartFederationToIdentityPool -> {
+                            StateResolution(FederatingToIdentityPool())
+                        }
+                        else -> defaultResolution
                     }
-                    is AuthenticationEvent.EventType.SignOutRequested -> {
-                        val action = authenticationActions.initiateSignOutAction(authenticationEvent, null)
-                        StateResolution(SigningOut(), listOf(action))
+                }
+                is FederatingToIdentityPool -> {
+                    when (val authorizationEvent = event.isAuthorizationEvent()) {
+                        is AuthorizationEvent.EventType.Fetched -> {
+                            StateResolution(FederatedToIdentityPool())
+                        }
+                        is AuthorizationEvent.EventType.ThrowError -> {
+                            StateResolution(Error(authorizationEvent.exception))
+                        }
+                        else -> defaultResolution
+                    }
+                }
+                is FederatedToIdentityPool -> {
```

Comment:
Looks like signOutRequested is not a valid event to be handled here: https://github.com/aws-amplify/amplify-android/pull/1986/files#diff-247378580a6f630b11f5d311bbf81a1fa4974acee7908083dadbd8f078216049R1371
Please remove if there is no valid transition to be done.

Reply:
I think I found why this transition exists, clearFederationToIdentityPool -> https://github.com/aws-amplify/amplify-android/pull/1986/files#diff-247378580a6f630b11f5d311bbf81a1fa4974acee7908083dadbd8f078216049R1740. Want to verify if my understanding is correct?

Reply:
Correct, after discussing with iOS, clearing federation is identical to sign out local, so it was decided to simplify states and remove clear federation states.